src/Tools/ExternalAccess/Razor/RazorProjectExtensions.cs

```diff
@@ -17,5 +17,14 @@ internal static class RazorProjectExtensions
         {
             return project.GetSourceGeneratorRunResultAsync(cancellationToken);
         }
+
+        internal static async ValueTask<IEnumerable<SourceGeneratedDocument>> GetSourceGeneratedDocumentsForGeneratorAsync(this Project project, string assemblyName, string assemblyPath, Version assemblyVersion, string typeName, CancellationToken cancellationToken)
+        {
+            var results = await project.GetSourceGeneratedDocumentsAsync(cancellationToken).ConfigureAwait(false);
+
+            var generatorIdentity = new SourceGeneratorIdentity(assemblyName, assemblyPath, assemblyVersion, typeName);
```

Comment:
`SourceGeneratorIdentity` isn't public, so have to pass through the requisite parts. We could make an IVT copy but doesn't seem worth it.